components/tfx/Evaluator/with_URI_IO/component.py

```diff
@@ -0,0 +1,103 @@
+    eval_config: {'JsonObject': {'data_type': 'proto:tensorflow_model_analysis.EvalConfig'}} = None,
+    feature_slicing_spec: {'JsonObject': {'data_type': 'proto:tfx.components.evaluator.FeatureSlicingSpec'}} = None,
+    fairness_indicator_thresholds: list = None,
+    beam_pipeline_args: list = None,
+) -> NamedTuple('Outputs', [
+    ('evaluation_uri', 'ModelEvaluationUri'),
+    #('blessing_uri', 'ModelBlessingUri'),
+]):
+    from tfx.components import Evaluator as component_class
+
+    #Generated code
+    import json
+    import os
+    import tempfile
+    import tensorflow
+    import uuid
+    from google.protobuf import json_format, message
+    from tfx.types import channel_utils, artifact_utils
+    from tfx.components.base import base_executor
+
+    arguments = locals().copy()
+
+    component_class_args = {}
+
+    for name, execution_parameter in component_class.SPEC_CLASS.PARAMETERS.items():
+        argument_value_obj = argument_value = arguments.get(name, None)
+        if argument_value is None:
+            continue
+        parameter_type = execution_parameter.type
+        if isinstance(parameter_type, type) and issubclass(parameter_type, message.Message): # Maybe FIX: execution_parameter.type can also be a tuple
+            argument_value_obj = parameter_type()
+            json_format.Parse(argument_value, argument_value_obj)
+        component_class_args[name] = argument_value_obj
+
+    for name, channel_parameter in component_class.SPEC_CLASS.INPUTS.items():
+        artifact_path = arguments.get(name + '_uri', None) or arguments.get(name + '_path', None)
+        if artifact_path:
+            artifact = channel_parameter.type()
+            artifact.uri = artifact_path + '/' # ?
+            if channel_parameter.type.PROPERTIES and 'split_names' in channel_parameter.type.PROPERTIES:
+                # Recovering splits
+                subdirs = tensorflow.io.gfile.listdir(artifact_path)
+                # Workaround for https://github.com/tensorflow/tensorflow/issues/39167
+                subdirs = [subdir.rstrip('/') for subdir in subdirs]
+                artifact.split_names = artifact_utils.encode_split_names(sorted(subdirs))
+            component_class_args[name] = channel_utils.as_channel([artifact])
+
+    component_class_instance = component_class(**component_class_args)
+
+    input_dict = {name: channel.get() for name, channel in component_class_instance.inputs.get_all().items()}
+    output_dict = {name: channel.get() for name, channel in component_class_instance.outputs.get_all().items()}
+    exec_properties = component_class_instance.exec_properties
+
+    # Generating paths for output artifacts
+    for name, artifacts in output_dict.items():
+        base_artifact_path = arguments.get('output_' + name + '_uri', None) or arguments.get(name + '_path', None)
+        if base_artifact_path:
+            # Are there still cases where output channel has multiple artifacts?
```

Comment:
> Are there still cases where output channel has multiple artifacts?

I believe for now there is not. Do you mind validating to see if we still need this logic?

Reply:
TFX still allows passing channels with multiple  artifacts and allows creating components that accept those multi-artifact channels. I'll verify the plans with Charles and Ruoyu. I'll probably remove this logic when TFX stops supporting it.